nautobot_ssot/integrations/aci/jobs.py

```diff
@@ -71,19 +64,45 @@ def data_mappings(cls):
 
     def load_source_adapter(self):
         """Method to instantiate and load the ACI adapter into `self.source_adapter`."""
-        self.source_adapter = AciAdapter(job=self, sync=self.sync, client=aci_creds[self.apic])
+        if not self.device_site:
+            self.logger.info("Device Location is unspecified so will revert to specified Controller's Location.")
+        _sg = self.apic.external_integration.secrets_group
+        username = _sg.get_secret_value(
+            access_type=SecretsGroupAccessTypeChoices.TYPE_HTTP,
+            secret_type=SecretsGroupSecretTypeChoices.TYPE_USERNAME,
+        )
+        password = _sg.get_secret_value(
+            access_type=SecretsGroupAccessTypeChoices.TYPE_HTTP,
+            secret_type=SecretsGroupSecretTypeChoices.TYPE_PASSWORD,
+        )
+        client = AciApi(
+            username=username,
+            password=password,
+            base_uri=self.apic.remote_url,
+            verify=self.apic.verify_ssl,
```

Comment:
I guess this should be taken from external integration, not apic directly?

Reply:
Also good catch. Fixed.